instrumentation/netty/netty-4.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/netty/v4_0/client/HttpClientResponseTracingHandler.java

```diff
@@ -100,9 +100,9 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     }
     if (msg instanceof HttpResponse) {
       HttpResponse httpResponse = (HttpResponse) msg;
-      span.setAttribute(SemanticAttributes.HTTP_STATUS_CODE, httpResponse.getStatus().code());
-      span.setStatus(
-          HttpStatusConverter.CLIENT.statusFromHttpStatus(httpResponse.getStatus().code()));
+      int code = httpResponse.getStatus().code();
+      span.setAttribute(SemanticAttributes.HTTP_STATUS_CODE, code);
+      span.setStatus(code >= 100 && code < 400 ? StatusCode.UNSET : StatusCode.ERROR);
```

Comment:
If the code is 200 even then will we set the status as UNSET?

Reply:
Yeah that's correct - we're duplicating this code from internally implemented APIs while we work on migrating our instrumentation to the OTEL instrumentation API.

This change helps us preserve existing behavior, but here's the relevant change in class visibility in OTEL: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5268/files#diff-2772ffa4248c0f9ab1a90b592d06c25e02e84be65303920b7f996f0739f3bfb2R25